controllers/webhook_cabundler.go

```diff
@@ -12,6 +18,64 @@ type k8s struct {
 	clientset kubernetes.Interface
 }
 
+//InitWebhookCertificates creates and injects req certs by the k8s webhooks
+func InitWebhookCertificates(certDir string) error {
+
+	secretName := os.Getenv("WEBHOOK_SECRET_NAME")
+	namespace := os.Getenv("SERVICE_NAMESPACE")
+	validatingWebhookConfigurationName := os.Getenv("VALIDATING_WEBHOOK_CONFIGURATION")
+	mutatingWebhookConfigurationName := os.Getenv("MUTATING_WEBHOOK_CONFIGURATION")
+	val := os.Getenv("CERTIFICATE_SANS")
+	dnsNames := strings.Split(val, ",")
+
+	if len(secretName) == 0 || len(namespace) == 0 || len(validatingWebhookConfigurationName) == 0 ||
+		len(mutatingWebhookConfigurationName) == 0 || len(dnsNames) == 0 {
+		os.Exit(1)
+		return errors.New("Need ENVS: WEBHOOK_SECRET_NAME, SERVICE_NAMESPACE, " +
+			"VALIDATING_WEBHOOK_CONFIGURATION, MUTATING_WEBHOOK_CONFIGURATION, CERTIFICATE_SANS")
+	}
+
+	rootCA, cert, key, err := generateCertificates(dnsNames)
+	if err != nil {
+		// Unable to create secret
+		return err
+	}
+
+	// Patching webhook secret
+	if err := patchWebhookSecret(rootCA.CAPem, cert, key, secretName, namespace); err != nil {
+		return err
+	}
+
+	// Patching validating webhook
+	if err := patchValidatingWebhookConfiguration(rootCA.CAPem, validatingWebhookConfigurationName); err != nil {
+		// setupLog.Error(err, "Unable to patch validating webhook")
+		return err
+	}
+
+	// Patching mutating webhook
+	if err := patchMutatingWebhookConfiguration(rootCA.CAPem, mutatingWebhookConfigurationName); err != nil {
+		return err
+	}
+
+	// Unable to create certDir
+	if err := os.MkdirAll(certDir, 0755); err != nil {
+		return err
+	}
+	// Unable to create ca.crt
+	if err := ioutil.WriteFile(filepath.Join(certDir, "ca.crt"), rootCA.CAPem, 0400); err != nil {
+		return err
+	}
+	// Unable to create tls.crt
+	if err := ioutil.WriteFile(filepath.Join(certDir, "tls.crt"), cert, 0400); err != nil {
+		return err
+	}
+	// Unable to create tls.key
+	if err := ioutil.WriteFile(filepath.Join(certDir, "tls.key"), key, 0400); err != nil {
+		return err
+	}
```

Comment:
What is the reason we're writing these to disk when we're already adding them to k8s secrets?

Reply:
The kubebuilder webhook server is setup to get the certificates it needs from a local path. See the docs on `CertDir`
```
// CertDir is the directory that contains the server key and certificate.
// if not set, webhook server would look up the server key and certificate in
// {TempDir}/k8s-webhook-server/serving-certs. The server key and certificate
// must be named tls.key and tls.crt, respectively.
CertDir string
```

Reply:
CertDir is used to create the controller manager. 
https://github.com/ForgeRock/secret-agent/blob/ec5da3c05e7e7695589b2b1939e58c4075920ebb/main.go#L70

Reply:
I see. Does that mean we're generating new certs every time the operator starts up? If so, do we want to consider only generating them if they don't exist?

Reply:
yes we generate them every time. We don't really do anything with the secret. we can probably  address this later. I don't think it would change functionality or security. Maybe I am missing something out.

Reply:
It will just increase startup time and CPU time unnecessarily and seems a bit dirty. I have created this ticket: https://github.com/ForgeRock/secret-agent/issues/28

Reply:
I do think we should make the dir 0750, and use something other than /tmp, otherwise I'll bet EntSec will complain.

Reply:
That's just the folder. The files inside are 400.
but sure, we can make it tighter

Reply:
that can be done as part of #28